pinot-plugins/pinot-batch-ingestion/pinot-batch-ingestion-common/src/main/java/org/apache/pinot/plugin/ingestion/batch/common/SegmentGenerationTaskRunner.java

```diff
@@ -167,19 +167,25 @@ private SegmentNameGenerator getSegmentNameGenerator(SegmentGeneratorConfig segm
         return new InputFileSegmentNameGenerator(segmentNameGeneratorConfigs.get(FILE_PATH_PATTERN),
             segmentNameGeneratorConfigs.get(SEGMENT_NAME_TEMPLATE), inputFileUri, appendUUIDToSegmentName);
       case BatchConfigProperties.SegmentNameGeneratorType.UPLOADED_REALTIME:
-        Preconditions.checkState(segmentGeneratorConfig.getCreationTime() != null,
-            "Creation time must be set for uploaded realtime segment name generator");
-        Preconditions.checkState(segmentGeneratorConfig.getUploadedSegmentPartitionId() != -1,
+        Preconditions.checkState(segmentNameGeneratorConfigs.get(BatchConfigProperties.SEGMENT_UPLOAD_TIME_MS) != null,
+            "Upload time must be set for uploaded realtime segment name generator");
+        Preconditions.checkState(segmentNameGeneratorConfigs.get(BatchConfigProperties.SEGMENT_PARTITION_ID) != null,
```

Comment:
How will the partitionId be provided by the task? The partitionId means the data is partitioned and same as stream that is ingesting to realtime table.

There should be a corresponding code in spark to partition data and derive id accordingly.

Reply:
Users need to partition data themselves. We have a spark job to replicate how stream ingestion partition data and put partitioned data in a path with partition id. e.g. `s3://.../partition=0/partitioned-data.parquet`
Then create segment task can get partition id from the path. So there will be one spark task per partition.